[Metricbeat] Remove elasticsearc.index.created from the SM code

sayden opened the pull request:
## What does this PR do?

SM `index` metricset does an API HTTP call to elasticsearch to retrieve index metadata information. This information is required to populate field `elasticsearch.index.created` or (`index_stats.created` if `xpack.enabled: true` when configuring the module).

The code removes the API call to `/_cluster/state/metadata`.

## Why is it important?

The HTTP call to `/_cluster/state/metadata` can be VERY expensive. The reason for this is that it returns the entire mapping of all indices on a cluster. If a user has many indices and / or each index has many fields, this could lead to very heavy responses from elasticsearchr (20 Mb in big clusters measured with curl).

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

* This is a breaking change, because users could be using field `elasticsearch.index.created` or `index_stats.created` and they will be removed.
* Stack Monitoring UI seems to keep working without any issues. I don't see this field populated anywhere.

elasticmachine commented:
## :green_heart: Build Succeeded


<!-- BUILD BADGES-->
> _the below badges are clickable and redirect to their specific view in the CI or DOCS_
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-25113/detail/PR-25113/1//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-25113/detail/PR-25113/1//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-25113/detail/PR-25113/1//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-25113/detail/PR-25113/1//artifacts) [![preview](https://img.shields.io/badge/docs-preview-yellowgreen)](http://beats_25113.docs-preview.app.elstc.co/diff)

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats


* Build Cause: `Pull request #25113 opened`



* Start Time: 2021-04-15T14:49:16.215+0000


* Duration: 106 min 37 sec


* Commit: 10f3b3df09a465161b6d587c3ee57e9dbb798f60



#### Test stats :test_tube:

| Test         | Results                         |
| ------------ | :-----------------------------: |
| Failed       | 0  |
| Passed       | 8339  |
| Skipped      | 2270 |
| Total        | 10609   |


#### Trends :test_tube:


![Image of Build Times](https://beats-ci.elastic.co/job/Beats/job/beats/job/PR-25113/buildTimeGraph/png)


![Image of Tests](https://beats-ci.elastic.co/job/Beats/job/beats/job/PR-25113/1/testReport/history/countGraph/png?start=0&amp;end=25)


</p>
</details>

<!-- TEST RESULTS IF ANY-->


<!-- STEPS ERRORS IF ANY -->












## :green_heart: Flaky test report
Tests succeeded.


<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

### Test stats :test_tube:
| Test         | Results                         |
| ------------ | :-----------------------------: |
| Failed       | 0  |
| Passed       | 8339  |
| Skipped      | 2270 |
| Total        | 10609   |



</p>
</details>



<!--COMMENT_GENERATED_WITH_ID_comment.id-->

ycombinator commented:
> Stack Monitoring UI seems to keep working without any issues. I don't see this field populated anywhere.

@sayden This would be my main concern with removing this field. Can we get someone from @elastic/stack-monitoring-ui team to **confirm** that removing this field will have no impact on the UI? Once we have that, I'm good with removing it to reduce the expensive API call.

jasonrhodes reviewed (approved):
I can confirm there are _no_ references to any field named "created" in the Stack Monitoring UI at this moment. 

For later reference, here is a search for "created" in x-pack/plugins/monitoring (excluding snapshot tests and SVG files):

<img width="895" alt="Screen Shot 2021-04-19 at 12 12 47 PM" src="https://user-images.githubusercontent.com/159370/115268822-aaa9d800-a108-11eb-9148-5c7107e1c597.png">


<details>
```
15 results - 10 files

x-pack/plugins/monitoring/public/plugin.ts:
   32  import { createLegacyAlertTypes } from './alerts/legacy_alert';
   33: import { createDiskUsageAlertType } from './alerts/disk_usage_alert';
   34  import { createThreadPoolRejectionsAlertType } from './alerts/thread_pool_rejections_alert';

  172      alertTypeRegistry.register(createCpuUsageAlertType());
  173:     alertTypeRegistry.register(createDiskUsageAlertType());
  174      alertTypeRegistry.register(createMemoryUsageAlertType());

x-pack/plugins/monitoring/public/alerts/disk_usage_alert/index.tsx:
  19  
  20: export function createDiskUsageAlertType(): AlertTypeModel<MonitoringAlertTypeParams> {
  21    return {

x-pack/plugins/monitoring/public/alerts/lib/alerts_toast.tsx:
  81          id="xpack.monitoring.healthCheck.disabledWatches.title"
  82:         defaultMessage="New alerts created"
  83        />

x-pack/plugins/monitoring/public/alerts/lib/get_alert_panels_by_category.test.tsx:
  48    params: {},
  49:   createdBy: null,
  50    updatedBy: null,
  51:   createdAt: new Date('2020-12-08'),
  52    updatedAt: new Date('2020-12-08'),

x-pack/plugins/monitoring/public/alerts/lib/get_alert_panels_by_node.test.tsx:
  46    params: {},
  47:   createdBy: null,
  48    updatedBy: null,
  49:   createdAt: new Date('2020-12-08'),
  50    updatedAt: new Date('2020-12-08'),

x-pack/plugins/monitoring/server/lib/mb_safe_query.ts:
  11   * queries/aggs continue to work but we need to handle the reality that these aliases will not
  12:  * exist for older metricbeat-* indices, created before the aliases existed.
  13   *

x-pack/plugins/monitoring/server/lib/details/get_series.js:
  245        console.log(
  246:         `metric.debug field=${metric.field} bucketsCreated: ${countBuckets(
  247            get(response, 'aggregations.check')

x-pack/plugins/monitoring/server/lib/elasticsearch/convert_metric_names.js:
  17   *
  18:  * Historically, the `get_nodes` function created an aggregation with multiple sub `date_histogram`
  19   * aggregations for each metric aggregation. From a top down view, the entire aggregations look liked:

x-pack/plugins/monitoring/server/lib/metrics/beats/metrics.js:
  72      description: i18n.translate('xpack.monitoring.metrics.beats.eventsRate.totalDescription', {
  73:       defaultMessage: 'All events newly created in the publishing pipeline',
  74      }),

x-pack/plugins/monitoring/server/routes/api/v1/alerts/enable.ts:
  79  
  80:         let createdAlerts: Array<SanitizedAlert<AlertTypeParams>> = [];
  81          const disabledWatcherClusterAlerts = await disableWatcherClusterAlerts(

  86          if (disabledWatcherClusterAlerts) {
  87:           createdAlerts = await Promise.all(
  88              alerts.map((alert) => alert.createIfDoesNotExist(alertsClient, actionsClient, actions))

  91  
  92:         return response.ok({ body: { createdAlerts, disabledWatcherClusterAlerts } });
  93        } catch (err) {
```

</details>

ycombinator commented on metricbeat/module/elasticsearch/index/data_xpack.go:
```diff
@@ -200,12 +194,6 @@ func addClusterStateFields(idx *Index, clusterState common.MapStr) error {
-	if err != nil {
-		return errors.Wrap(err, "failed to get index creation time")
-	}
-	idx.Created = created
```
I think we can remove the `Created` field from the struct as well?

sayden replied:
Good point

ycombinator commented:
Thanks for checking and confirming, @jasonrhodes.

ycombinator commented:
> The code removes the API call to /_cluster/state/metadata.

@sayden Reviewing the code changes in this PR, AFAICT we are not removing the API call itself. This PR is simply removing the code that _parses_ the cluster state API response to extract the index metadata field from it.

sayden commented:
> > The code removes the API call to /_cluster/state/metadata.
> 
> @sayden Reviewing the code changes in this PR, it looks like we are not actually removing the API call to ES. This PR is simply removing the code that _parses_ the cluster state API response to extract the index metadata field from it.

Yes, sorry for that. I cherry-picked the commit from a different branch and I didn't realize that I split the change in two commits

sayden commented:
Ok, I actually removed the call now :sweat_smile: (actually, the request into a different call), deleted the field in struct and added a changelog entry into breaking changes.

ycombinator reviewed (approved):
LGTM.

sayden commented:
Thank you for your help too, @jasonrhodes

elasticmachine commented:
Pinging @elastic/integrations-services (Team:Services)
